src/components/common/TextWithLink.tsx

```diff
@@ -0,0 +1,31 @@
+export interface TextWithLinkProps {
+  className?: string;
+  href?: string;
+  onClick?: () => void;
+  text: string;
+}
+
+/**
+ * Renders a string with an embedded link.
+ *
+ * @param className
+ * @param text The text to render. The link should be surrounded by square brackets. For
+ * example, `'See [Jane] run'` would be rendered as `See <a href=...>Jane</a> run`.
+ */
+export default function TextWithLink({ className, href, onClick, text }: TextWithLinkProps): JSX.Element {
+  const linkStart = text.indexOf('[');
+  const linkEnd = text.indexOf(']');
+  const prefix = text.substring(0, linkStart);
+  const linkText = text.substring(linkStart + 1, linkEnd);
+  const suffix = text.substring(linkEnd + 1);
+
```

Comment:
if there's no linkText, return input text as-is?
guessing visually the current code won't show any link if there's no link text .. so maybe it's ok..

Reply:
Good catch. I think in that case it makes more sense to turn the whole string into a link. Otherwise a bad translation could make it impossible to follow the link. Updated the PR to do that.

Reply:
That's a good idea!! This is definitely better.